drivers/include/vl53l1x.h

```diff
@@ -0,0 +1,586 @@
+
+/** to be able to use also VL53L1_ instead of VL53L1X_ for consistency */
+#define VL53L1X_ VL53L1_
+#define vl53l1x_ vl53l1_
+/** platform specific error codes are starting at VL53L1X_ERROR_BASE */
+#define VL53L1X_ERROR_BASE          VL53L1_ERROR_PLATFORM_SPECIFIC_START
+
+#else /* MODULE_VL53L1X_ST_API */
+
+#define VL53L1X_ERROR_BASE          (0) /**< base value of driver error codes */
+
+#define VL53L1X_RANGESTATUS_RANGE_VALID                    0   /**< range is valid */
+#define VL53L1X_RANGESTATUS_SIGMA_FAIL                     1   /**< sigma failure */
+#define VL53L1X_RANGESTATUS_SIGNAL_FAIL                    2   /**< signal failure */
+#define VL53L1X_RANGESTATUS_RANGE_VALID_MIN_RANGE_CLIPPED  3   /**< target is below minimum
+                                                                   detection threshold */
+#define VL53L1X_RANGESTATUS_OUTOFBOUNDS_FAIL               4   /**< phase out of valid limits,
+                                                                    different to a wrap exit. */
+#define VL53L1X_RANGESTATUS_HARDWARE_FAIL                  5   /**< hardware failure */
+#define VL53L1X_RANGESTATUS_RANGE_VALID_NO_WRAP_CHECK_FAIL 6   /**< range is valid but the wrap
+                                                                    around check has not been
+                                                                    done */
+#define VL53L1X_RANGESTATUS_WRAP_TARGET_FAIL               7   /**< wrapped target, no matching
+                                                                    phase in other VCSEL period
+                                                                    timing */
+#define VL53L1X_RANGESTATUS_XTALK_SIGNAL_FAIL              9   /**< specific to lite ranging */
+#define VL53L1X_RANGESTATUS_SYNCRONISATION_INT             10  /**< first interrupt when starting
+                                                                    ranging in back to back mode */
+#define VL53L1X_RANGESTATUS_MIN_RANGE_FAIL                 13  /**< user ROI input is not valid */
+#define VL53L1X_RANGESTATUS_NONE                           255 /**< No Update. */
+
+#endif /* MODULE_VL53L1X_ST_API */
+
+/**
+ * @name Definition of error codes
+ * @{
+ */
+#define VL53L1X_OK              (0)                      /**< success */
+#define VL53L1X_ERROR_I2C       (VL53L1X_ERROR_BASE - 1) /**< I2C communication error */
+#define VL53L1X_ERROR_WRONG_ID  (VL53L1X_ERROR_BASE - 2) /**< wrong id read */
+#define VL53L1X_ERROR_NO_DATA   (VL53L1X_ERROR_BASE - 3) /**< no data available */
+#define VL53L1X_ERROR_RAW_DATA  (VL53L1X_ERROR_BASE - 4) /**< read raw data failed */
+#define VL53L1X_ERROR_NO_PIN    (VL53L1X_ERROR_BASE - 5) /**< pin not defined */
+#define VL53L1X_ERROR_TIMEOUT   (VL53L1X_ERROR_BASE - 6) /**< timeout happened */
+#define VL53L1X_ERROR_INV_ARG   (VL53L1X_ERROR_BASE - 7) /**< invalid argument */
+#define VL53L1X_ERROR_INV_PARAM (VL53L1X_ERROR_BASE - 8) /**< invalid params */
```

Comment:
> Otherwise, it is now preferred to return 0 on success and negative errno codes on failure. 

I know, I was involved in this discussion some years ago in issue #10686. The use of POSIX error codes is not trivial for drivers, since it is often not clear which error should be assigned to which POSIX error code.

In the case of this driver, it is even more difficult because the original error codes returned by functions of the ST VL53L1X package are returned directly by the driver functions. It would be difficult to convert them, especially if a number of functions of the ST VL53L1X package are called in a single driver function: